### PR TITLE
rejected thumb css improvement

### PR DESCRIPTION
**Pull request by Nilvus**
A really simple CSS update, especially to complete rejected thumbnail work from @AlicVB.

### CODE REVIEW

**Comment by TurboGit** [comment]
@Nilvus : Can you describe what this is doing? On my side I see no difference. We have been speaking about making the whole thumb background dimmer (as already done for the image) to make the rejected thumbs stand out more. Is that what this PR try to achieve?

File: data/themes/darktable.css [hunk 1]
```diff
@@ -2525,6 +2520,22 @@ Details :
...
+}
+
+.dt_thumbnail_rating_6 #thumb_ext,
+.dt_thumbnail_rating_6 #thumb_altered,
+.dt_thumbnail_rating_6 .dt_thumb_btn,
+.dt_thumbnail_rating_6#thumb_main:hover #thumb_ext,
+.dt_thumbnail_rating_6#thumb_main:hover #thumb_altered,
+.dt_thumbnail_rating_6#thumb_main:hover .dt_thumb_btn
+{
+  color: rgba(0,0,0,0.2); /* only the transparency is used to fade the image drawing */
```
Comment (AlicVB) [inline data/themes/darktable.css:2536]: I fear that this feature is only implemented for #thumb_image... So this should have no real effect

**Review by AlicVB — CHANGES_REQUESTED** [review]
not tested yet, but looking at the code, I'm not sure this will change anything...

**Comment by TurboGit** [comment]
> not tested yet, but looking at the code, I'm not sure this will change anything...

And that's what I have experienced on my side :)

**Comment by Nilvus** [comment]
I'm quite surprised about your reactions but yes, of course that change something (I check that and see changes on my side, so maybe need indeed some explanations. This change how extension name and icons (altered, group and stars one) are displayed. Said that, this let me think that I forgot to test all overlays modes, so maybe that could explain your reactions.

File: data/themes/darktable.css (hunk 1 above)
Comment (Nilvus) [reply data/themes/darktable.css:2536]: No, it's not only for #thumb_image as you linked dt_thumbnail_rating_6 id on thumb_main, so all thumbnail and all part, thumb_image and stars, extension name, etc. are inside thumb_main widget.

Comment (AlicVB) [reply data/themes/darktable.css:2538]: sorry, I see that I've half explained what I have in mind : 
What is not implemented is the transparency thing. iirc only the thumbnail image itself is able to handle the part "only the transparency is used..." as it's some specific cairo code added when drawing the image in the canvas...
For the rest of more "classic" css you can of course use that...

**Comment by Nilvus** [comment]
To precise: the idea here is to dim all thumbnail part, so text, stars also (except of course rejected red cross). So you could see that this apply effect on extension name, group and altered icon but also on stars (when hovering the thumbnail especially) and now text.

Anyway, I just see that how I code that create some glitches on some overlays. Will try to improve that. Hope you will now better understand the purpose.

**Comment by Nilvus** [comment]
@AlicVB: I just a bug related to your PR about that. Just set overlay mode to no overlay (first mode). Quit darktable and relaunch it. You could see that rejected images do not have dimmer effect. Change filter (all or rejected ones, etc.) and see no changes. Change overlay mode, see effect applied on rejected images then go back to no overlay mode, effect is now correctly applied.

If you want me to open an issue, just ask!

**Comment by Nilvus** [comment]
Should be good now.

Before (example on overlays mode 6):

image

After:

image

File: data/themes/darktable.css (hunk 1 above)
Comment (Nilvus) [reply data/themes/darktable.css]: For text, it's not necessary to use cairo as CSS is applied directly and naturally on all "text"

Comment (Nilvus) [reply data/themes/darktable.css]: To precise: it's only here color changes so all items with already color set could be set with that. What you add was a possibility to set that with CSS on image but it's already there for other parts.

Comment (AlicVB) [reply data/themes/darktable.css]: yes, for text and icons, here, you define the whole color, not only the transparency (just test `rgba(255,0,0,0.2);` to see)
now I wonder if there's not something in css to just add transparency to an existing color... maybe you know that better than me !

**Comment by AlicVB** [comment]
@Nilvus : I've justed, and it works great. As said earlier, do you know if there's a way in css to change only the transparency of a color ? That would be better and avoid breaking themes where the default color is not black...

For your issue, I'll have a look asap. Just annoy me if I forget, as usual ;)

File: data/themes/darktable.css (hunk 1 above)
Comment (Nilvus) [reply data/themes/darktable.css]: Of course

**Comment by Nilvus** [comment]
> @Nilvus : I've justed, and it works great. As said earlier, do you know if there's a way in css to change only the transparency of a color ? That would be better and avoid breaking themes where the default color is not black...

There's no direct possibility here. Not really CSS issue but Gtk limit with CSS. Not all CSS properties are implemented or completely implemented. There's opacity setting but that doesn't work here. Anyway, if you set a color, to be sure transparency is correctly applied on, you need to precise color to apply transparency. CSS works also a lot on parent/chlidren behavior. That means that to replace an already existing setting, you need to set replacement with same similar things and just update setting you want. That works a lot for example on tag and id in CSS. As you can see here on my code, I need to precise a lot of things, like for example the .dt_thumbnail_rating_6#thumb_main:hover... .dt_thumbnail_rating_6 is the same widget than thumb_main but as thumb_main hover is set previously in all items (and not only dt_thumbnail_rating_6), it's necessary to precise all those to be sure that CSS is applied (just remove #thumb_main to check, CSS will not be applied). Hope I'm clear here but it's an important thing in CSS.

To be simple, what your asked for is a good idea but not possible.

**Comment by AlicVB** [comment]
I was sure you will have better answer than me :)

all good for me, then. Thanks !

**Comment by TurboGit** [comment]
> To precise: the idea here is to dim all thumbnail part, so text, stars also (except of course rejected red cross). So you could see that this apply effect on extension name, group and altered icon but also on stars (when hovering the thumbnail especially) and now text.

What about the thumb background? Having it deemed will help a lot seeing where are the rejected pictures.

**Review by TurboGit — APPROVED** [review]
Thanks, this goes in the right direction so I'll merge now. If it is possible to also dim the whole thumb background it would be very nice.

**Pull request merged**